### ISSUE TITLE
日服生息演算自动刷取异常

### ISSUE THREAD

**Issue by WanKafka** [MaaAssistantArknights/MaaAssistantArknights#6319]
### 在提问之前...

- [X] 我理解 Issue 是用于反馈和解决问题的，而非吐槽评论区，将尽可能提供更多信息帮助问题解决
- [X] 我填写了简短且清晰明确的标题，以便开发者在翻阅 issue 列表时能快速确定大致问题。而不是“一个建议”、“卡住了”等
- [X] 我已查看最新测试版本的更新内容，并未提及该 bug 已被修复的情况

### 问题描述

卡在如图所示的地方

### 日志文件

[asst.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12578240/asst.txt)
[Uploading gui.log…]()
麻烦大佬把第一个txt文件改成rar，不压缩没法上传，压缩了文件格式不支持，只能改掉了。
时间在11号下午四点-四点半之间

### 截图

![N{8 E685M%K0ZYY4A_G8P13](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/104065679/95ca153b-ccd8-48da-a8a4-eb5ec6d42a12)

### 还有别的吗？

mumu12 win10

**Comment by Constrat** [comment]
It's as @THSLP13 said. I don't have access to any JP accounts. I've only adapted YostarEN as of latest version.

**Comment by THSLP13** [comment]
> It's as @THSLP13 said. I don't have access to any JP accounts. I've only adapted YostarEN as of latest version.

Maybe I can help...If i have time to do this.

**Comment by THSLP13** [comment]
It seems like OCR problems at
`[2023-09-12 14:17:34.851][TRC][Px9be8][Tx9b54] asst::WordOcr [{ 瓦: [ 258, 48, 45, 33 ], score: 0.369259 }, { 賛皛エuア: [ 342, 52, 66, 10 ], score: 0.645931 }, { ？？?: [ 703, 55, 27, 10 ], score: 0.828924 }, { 壁回山亡進2α所護: [ 343, 62, 64, 10 ], score: 0.409299 }, { _-: [ 309, 49, 28, 22 ], score: 0.383626 }, { 狩猟エリア: [ 517, 68, 69, 11 ], score: 0.982442 }, { 閲囲雷: [ 704, 62, 32, 13 ], score: 0.656750 }, { 4: [ 489, 72, 22, 8 ], score: 0.362690 }, { …: [ 490, 80, 13, 7 ], score: 0.766233 }, { 門■nn置znn: [ 516, 79, 62, 12 ], score: 0.694357 }, { 1nrc: [ 2, 98, 32, 10 ], score: 0.839591 }, { ?？？: [ 3, 86, 28, 11 ], score: 0.962691 }, { 〆s: [ 201, 105, 17, 14 ], score: 0.293217 }, { 狩猟エリア: [ 226, 102, 71, 11 ], score: 0.999401 }, { :e: [ 153, 116, 18, 10 ], score: 0.495103 }, { —■門■再室■■: [ 227, 116, 59, 7 ], score: 0.713391 }, { >: [ 558, 188, 25, 24 ], score: 0.557676 }, { 斉源工アア: [ 644, 183, 72, 10 ], score: 0.464292 }, { .: [ 621, 195, 7, 5 ], score: 0.299841 }, { R空■_rcEZ■n: [ 646, 195, 71, 10 ], score: 0.600480 }, { ど: [ 118, 240, 20, 16 ], score: 0.403012 }, { 荊猟エリア: [ 145, 235, 78, 16 ], score: 0.869242 }, { /: [ 60, 250, 35, 18 ], score: 0.363220 }, { H山m■■zno: [ 146, 251, 67, 10 ], score: 0.474944 }, { -:: [ 237, 256, 16, 12 ], score: 0.935426 }, { 記屋: [ 321, 266, 51, 14 ], score: 0.573067 }, { アート: [ 326, 302, 55, 13 ], score: 0.482268 }, { 5: [ 566, 294, 54, 53 ], score: 0.262924 }, { 〆: [ 634, 307, 20, 18 ], score: 0.437941 }, { 資源エリア: [ 662, 304, 75, 13 ], score: 0.907073 }, { 口: [ 264, 315, 9, 8 ], score: 0.999067 }, { R塵■LrEZ[: [ 664, 319, 72, 11 ], score: 0.544759 }, { 回: [ 104, 393, 55, 47 ], score: 0.935506 }, { 〆s: [ 172, 400, 24, 23 ], score: 0.408599 }, { 遭遇: [ 204, 397, 34, 15 ], score: 0.999976 }, { Enc■unt■r: [ 204, 415, 57, 11 ], score: 0.799863 }, { 回: [ 465, 437, 50, 44 ], score: 0.800308 }, { s〆<: [ 531, 443, 24, 24 ], score: 0.294324 }, { 遭遇: [ 561, 440, 37, 17 ], score: 0.999998 }, { 瓦: [ 240, 466, 61, 57 ], score: 0.613557 }, { Eno■LnL■: [ 564, 459, 57, 11 ], score: 0.699498 }, { 〆s: [ 311, 476, 25, 24 ], score: 0.385052 }, { 資源エリア: [ 344, 474, 84, 17 ], score: 0.996053 }, { R座ロLrEEZcn: [ 344, 492, 84, 12 ], score: 0.707506 }] by OCR Pipeline , cost 718 ms`
It matches HINT` "Reclamation@ClickAnyZoneOcr": {
        "algorithm": "OcrDetect",
        "cache": false,
        "roi": [
            308,
            73,
            839,
            567
        ],
        "text": [
            "資源",
            "狩猟"
        ],
        "postDelay": 1500,
                "action": "ClickSelf"
    },`
But It stucks here and return to main page then restart the alogrithm.

**Comment by Constrat** [comment]
What needs to be done is adding the JP texts. For that screen you need do add this.
![Screenshot_20230912-082430__01](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/56174894/f9a3142a-5338-44bf-9308-52627f86bd11)

**Comment by THSLP13** [comment]
> What needs to be done is adding the JP texts. For that screen you need do add this. ![Screenshot_20230912-082430__01](https://user-images.githubusercontent.com/56174894/267230760-f9a3142a-5338-44bf-9308-52627f86bd11.jpg) But this is just the start. I suggest you going in tools/OverseasClient and run the missing jsontranslate python file it will show you what tasks need to be translated.

Thanks.

**Comment by THSLP13** [comment]
But it always stuck here
![Screenshot 09-12 15H08M](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/62548119/218dca04-e1af-4f95-97ad-483d0f73c38a)

This is my current task.json for JP (Translating in Progress)
[task.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12583431/task.txt)

**Comment by THSLP13** [comment]
> But it always stuck here ![Screenshot 09-12 15H08M](https://user-images.githubusercontent.com/62548119/267242479-218dca04-e1af-4f95-97ad-483d0f73c38a.png)
> 
> This is my current task.json for JP (Translating in Progress) [task.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12583431/task.txt)

When it goes there,after 1 minute it backs to menu then retreat.Then start the next cycle of the task.
Just loop and loop.

**Comment by Constrat** [comment]
Don't modify that task (ClickAnyZoneOcr). With the main tasks.json it already searches for "Zone". If you want you can just add "Zone","エリア"

**Comment by THSLP13** [comment]
@Constrat 
Translation almost done!
But Recognization in Map is still not working.
Here is my current task.json 
[task.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12588053/task.txt)
Almost success but fails at `Reclamation@ClickAnyZoneOcr`
It always timeout though OCR find proper results.

**Comment by THSLP13** [comment]
This is the last log of `Reclamation@ClickAnyZoneOcr`

Here is the detail.
`[2023-09-12 23:12:31.248][INF][Px23b0c][Tx23e70] {"class":"asst::ProcessTask","details":{"cur_retry":20,"retry_times":20,"to_be_recognized":["ClickAnyZone@Reclamation@Begin"]},"first":["ClickAnyZone@Reclamation@Begin"],"pre_task":"","subtask":"ProcessTask","taskchain":"ReclamationAlgorithm","taskid":1}
[2023-09-12 23:12:31.517][INF][Px23b0c][Tx23e70] Call ` D:\Ark\adb\adb.exe -s 127.0.0.1:5565 exec-out "screencap | gzip -1" ` ret 0 , cost 268 ms , stdout size: 1034327 , socket size: 0
[2023-09-12 23:12:31.543][TRC][Px23b0c][Tx23e70] asst::PipelineAnalyzer::analyze ClickAnyZone@Reclamation@Begin
[2023-09-12 23:12:32.020][TRC][Px23b0c][Tx23e70] asst::WordOcr [{ 狩猟エリア: [ 372, 57, 70, 13 ], score: 0.999879 }, { ？？？: [ 655, 63, 28, 13 ], score: 0.923164 }, { Hni■z■n: [ 374, 69, 57, 10 ], score: 0.776417 }, { ★: [ 296, 64, 31, 19 ], score: 0.199795 }, { N画1nr-E: [ 654, 76, 36, 10 ], score: 0.648067 }, { 正: [ 44, 146, 50, 38 ], score: 0.473700 }, { 〆: [ 108, 152, 19, 18 ], score: 0.600599 }, { 遠征: [ 134, 149, 33, 14 ], score: 0.926566 }, { sコtan: [ 136, 163, 41, 10 ], score: 0.725448 }, { ~: [ 494, 169, 24, 22 ], score: 0.810715 }, { 資原エリア: [ 573, 162, 74, 16 ], score: 0.962861 }, { P医S回LFE置Z■n: [ 575, 178, 70, 10 ], score: 0.620962 }, { 記巨: [ 271, 277, 50, 20 ], score: 0.612579 }, { E: [ 199, 318, 7, 7 ], score: 0.349036 }, { 9全ー5: [ 254, 315, 71, 14 ], score: 0.249453 }, { 万: [ 25, 379, 43, 39 ], score: 0.353042 }, { ~: [ 498, 375, 20, 16 ], score: 0.556528 }, { 資源エリア: [ 584, 377, 80, 13 ], score: 0.996206 }, { 資源エリア: [ 119, 381, 82, 17 ], score: 0.999986 }, { 〆: [ 88, 384, 24, 21 ], score: 0.501449 }, { s手: [ 554, 390, 21, 11 ], score: 0.166483 }, { R医■urEEZ■n最: [ 584, 391, 82, 13 ], score: 0.713191 }, { &x: [ 484, 376, 42, 39 ], score: 0.422272 }, { 同業■門置zn: [ 119, 397, 78, 12 ], score: 0.513186 }, { 口: [ 239, 492, 60, 53 ], score: 0.951345 }, { 遭遇: [ 344, 498, 37, 17 ], score: 0.999992 }, { 94: [ 314, 514, 16, 13 ], score: 0.419683 }, { EnoumitEr: [ 345, 516, 59, 14 ], score: 0.727360 }] by OCR Pipeline , cost 476 ms
[2023-09-12 23:12:32.021][TRC][Px23b0c][Tx23e70] Proced []
[2023-09-12 23:12:32.021][TRC][Px23b0c][Tx23e70] asst::ProcessTask::_run | leave, 773 ms
[2023-09-12 23:12:32.021][TRC][Px23b0c][Tx23e70] ready to sleep 500
[2023-09-12 23:12:32.523][TRC][Px23b0c][Tx23e70] end of sleep 500
[2023-09-12 23:12:32.523][INF][Px23b0c][Tx23e70] Assistant::append_callback | SubTaskError {"class":"asst::ProcessTask","details":{},"first":["ClickAnyZone@Reclamation@Begin"],"pre_task":"","subtask":"ProcessTask","taskchain":"ReclamationAlgorithm","taskid":1,"uuid":"cefd494079a4f755"}
[2023-09-12 23:12:32.523][TRC][Px23b0c][Tx23e70] asst::ProcessTask::on_run_fails | enter
[2023-09-12 23:12:32.523][TRC][Px23b0c][Tx23e70] asst::ProcessTask::on_run_fails | leave, 0 ms`

**Comment by Constrat** [comment]
> Almost success but fails at `Reclamation@ClickAnyZoneOcr`
> It always timeout though OCR find proper results.

Ok so this is a bit complicated but: if you look at the logs, you will see that the errors happens on `ClickAnyZone@Reclamation@Begin`. But. that task has no `text` field. Because of this it uses the CN version of `Reclamation@Begin` because the task is used as template. To fix this you just need to add
<details><summary>ClickAnyZone@Reclamation@Begin</summary>
<p>

```
    "ClickAnyZone@Reclamation@Begin": {
        "baseTask": "Reclamation@ClickAnyZoneOcr",
        "next": [
            "Stop"
        ]
    },
```
</p>
</details> 
to the JP tasks.json. This should fix everything.

**Comment by THSLP13** [comment]
> 

Thanks,I think I just don't know well about how exactly it works.

**Comment by THSLP13** [comment]
@WanKafka 
你好，现在我电脑上进行的本地测试已经可以顺利完成一轮的生息演算
但是翻译还差点东西，所以我暂时无法提交文件让这份翻译后配置文件正式生效

Translate is almost done and MAA in my PC now can independently finish one cycle of Reclamation Alogrithm.
I'm now finding about the position of rest missing config translation such as "Evacuate with Remaining Items"
...This is because difference between in-game text and my translation result.

**Comment by WanKafka** [comment]
> @WanKafka 你好，现在我电脑上进行的本地测试已经可以顺利完成一轮的生息演算 但是翻译还差点东西，所以我暂时无法提交文件让这份翻译后配置文件正式生效
> 
> Translate is almost done and MAA in my PC now can independently finish one cycle of Reclamation Alogrithm. I'm now finding about the position of rest missing config translation such as "Evacuate with Remaining Items" ...This is because difference between in-game text and my translation result.

**Comment by WanKafka** [comment]
> @WanKafka 你好，现在我电脑上进行的本地测试已经可以顺利完成一轮的生息演算 但是翻译还差点东西，所以我暂时无法提交文件让这份翻译后配置文件正式生效
> 
> Translate is almost done and MAA in my PC now can independently finish one cycle of Reclamation Alogrithm. I'm now finding about the position of rest missing config translation such as "Evacuate with Remaining Items" ...This is because difference between in-game text and my translation result.

**Comment by WanKafka** [comment]
> > @WanKafka 你好，现在我电脑上进行的本地测试已经可以顺利完成一轮的生息演算 但是翻译还差点东西，所以我暂时无法提交文件让这份翻译后配置文件正式生效
> > Translate is almost done and MAA in my PC now can independently finish one cycle of Reclamation Alogrithm. I'm now finding about the position of rest missing config translation such as "Evacuate with Remaining Items" ...This is because difference between in-game text and my translation result.

感谢大佬，期待您的进一步优化

**Comment by THSLP13** [comment]
@WanKafka 

如果你实在着急的话，就把跨域通讯塔拆了然后用我的配置（？？？
如果你有撤退并带走物资的图片请请请务必发在这里

[task.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12592223/task.txt)

**Comment by Manicsteiner** [comment]
> 

[task.txt](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12592934/task.txt)
添加撤退時的物資確認。修正一些应使用日文汉字的文本。

**Comment by THSLP13** [comment]
@Manicsteiner 
现在json文件已经可以正常运作，是否需要发起pull request

**Comment by Constrat** [comment]
Fixed in 2ca8d8c
